purge_olean.sh

```diff
@@ -1,13 +1,20 @@
-#! /bin/sh
-# olean=./src/tactic/ring.olean
-# lean_file=`echo $$olean_file | sed "s/\\.olean/.lean/"`
+#!/bin/sh
 
-for olean_file in `find . -name "*.olean"`;
+# Top-level directory of the Git repository
+dir=`git rev-parse --show-toplevel`
+
+# Delete every <path>.olean without a matching <path>.lean.
+
+for olean_file in `find $dir -name "*.olean"`
 do
-    echo olean file: $olean_file
     lean_file=`echo $olean_file | sed "s/\.olean/.lean/"`
     if [ ! -e $lean_file ]; then
-        echo "absent " $lean_file;
+        echo "rm $olean_file"
         rm $olean_file
     fi
 done
+
+# Delete all empty directories. An empty directory may have been created if it
+# does not contain any .lean files and all of its .olean files were deleted.
+
+find $dir -type d -empty -delete
```

Comment:
This will also delete empty subdirectories of `.git` which I'm not sure is a good idea. How about
```suggestion
find $dir -type d -not -path '*/.git/*' -empty -delete
```

Reply:
@rwbarton Good suggestion. I went with the approach of conservatively including only the `src` and `test` directories in 360e8e2. I don't believe we will be find `.olean` files anywhere else, so this should be safe.